### PR TITLE
Dip-178-P2M-Refund

### PR DESCRIPTION
**Pull request by YinonFirstDAG**
Please review the first draft of DIP-178 about P2M Refund

### CODE REVIEW

File: dips/dip-158.md [hunk 1]
```diff
@@ -1,42 +1,36 @@
 ---
 dip: 158
-title: P2M (eCommerce) Payments Off-Chain Protocol
+title: P2M Charge Payment - Off-Chain Protocol
```
Comment (dimroc) [inline dips/dip-158.md:3]: This PR mistakenly includes DIP 158: P2M Charge Payment. Let's cut this file.

File: dips/dip-178.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+**Reminder:** Successful refund from the business to the customer requires a previous payment from the customer to the merchant. 
+
+### Successful Refund Sequence
+| Step | Command Type | Triggered by | Description |
+|-|-|-|-|
+|1|InitRefundPayment|Merchant VASP|Merchant VASP initializes the refund. Customer VASP will respond it is willing to accept the refund|
+|2|RefundReadyForSettlementNotification|Merchant VASP|The merchant VASP indicates the refund is ready for settlement. The VASP is expected to submit a transaction on-chain immediately before or after this step|
+
+#### Refund Sequence Diagram
+![Refund Sequence Diagram](https://static.swimlanes.io/b3f04660b9617526e07cd41cdb5ee8c1.png)
```
Comment (dimroc) [inline dips/dip-178.md:85]: I would place the `Refund Sequence Diagram` section higher up, above `Command Types`

File: dips/dip-178.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+## Command Types
+The following is a list of new and existing values for the `command_type` field of the `CommandRequestObject`. Each command type denotes a step in the refund payment off-chain negotiations and specifies the data of the requests and the response 
+* `InitRefundPayment` - Initializes a refund payment
+* `RefundReadyForSettlementNotification` - Notifies the payment is ready for settlement on-chain
+* `AbortRefund` - Aborts a payment
+
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|[InitRefundPayment](#initrefundpayment)|Merchant VASP > Customer VASP|This functionality allows the merchant VASP to init the refund. For example, this may occur after the merchant requested its VASP to refund the customer due to some items being out of stock|Reference ID ; Refund ID ; (optional) Amount|(optional) Recipient Signature|
```
Comment (dimroc) [inline dips/dip-178.md:67]: Why do we need a Refund ID in addition to the Reference ID?

If we do need it, can we include a snippet explaining its purpose?

File: dips/dip-178.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+---
+This section describes the expected sequences in P2M refund scenario.
+
+**Reminder:** Successful refund from the business to the customer requires a previous payment from the customer to the merchant. 
+
+### Successful Refund Sequence
+| Step | Command Type | Triggered by | Description |
+|-|-|-|-|
+|1|InitRefundPayment|Merchant VASP|Merchant VASP initializes the refund. Customer VASP will respond it is willing to accept the refund|
+|2|RefundReadyForSettlementNotification|Merchant VASP|The merchant VASP indicates the refund is ready for settlement. The VASP is expected to submit a transaction on-chain immediately before or after this step|
```
Comment (dimroc) [inline dips/dip-178.md:82]: If we don't need Refund ID, can we reuse the original `ReadyForSettlementNotification`?

**Review by dimroc — CHANGES_REQUESTED** [review]

**Comment by YinonFirstDAG** [comment]
Will open a new PR after DIP-158 changes are merged to main

File: dips/dip-158.md (hunk 1 above)
Comment (YinonFirstDAG) [reply dips/dip-158.md:3]: Closed this PR.
Will open a new one after DIP-158 changes is merged to main.

File: dips/dip-178.md (hunk 3 above)
Comment (YinonFirstDAG) [reply dips/dip-178.md:67]: There could be multiple refunds for the same payment.
In order to identify a specific refund and maintain a successive protocol, refund ID must be passed

File: dips/dip-178.md (hunk 4 above)
Comment (YinonFirstDAG) [reply dips/dip-178.md:82]: Since we need Refund ID we need a new command that allows refund ID as input.

**Pull request closed without merging**